impeller/scene/importer/vertices_builder.h

```diff
@@ -35,26 +27,48 @@ class VerticesBuilder {
     kFloat,
   };
 
-  VerticesBuilder();
-
-  void WriteFBVertices(std::vector<fb::Vertex>& vertices) const;
+  enum class AttributeType {
+    kPosition,
+    kNormal,
+    kTangent,
+    kTextureCoords,
+    kColor,
+  };
 
-  void SetAttributeFromBuffer(Attribute attribute,
-                              ComponentType component_type,
-                              const void* buffer_start,
-                              size_t stride_bytes,
-                              size_t count);
+  using ComponentConverter =
+      std::function<Scalar(const void* source, size_t byte_offset)>;
+  struct ComponentProperties {
+    size_t size_bytes = 0;
+    ComponentConverter convert_proc;
+  };
 
- private:
+  struct AttributeProperties;
+  using AttributeWriter =
+      std::function<void(Scalar* destination,
+                         const void* source,
+                         const ComponentProperties& component_props,
+                         const AttributeProperties& attribute_props)>;
   struct AttributeProperties {
```

Comment:
Just noticed this, but, can we zero these out by default?

Reply:
Done. I forget to do this a lot, I wonder if there's a clang-format lint that can help.